lib/core/treenode.js

```diff
@@ -18,6 +18,10 @@ class TreeNode {
     return this.__instance;
   }
 
+  get redact() {
+    return this.options ? this.options.redact || false : false;
```

Comment:
why not simply say `this.options && this.options.redact`?

Reply:
that might return undefined, which works in this context but here it explicitly returns a boolean

Reply:
Not really, It returns something or false,  and it's also a very confusing statement. `this.options` is always going to be defined so you could say `!!this.options.redact` to to always get a boolean. Or just `this.options.redact` is also fine.

Reply:
sure will check and modify it to make it more clear